examples/CPMPY_Pred+Opt.ipynb

```diff
@@ -0,0 +1,906 @@
+    "data = np.load(io.BytesIO(response.content))\n",
+    "x_train,  x_test, y_train,y_test = data['X_1gtrain'],data['X_1gtest'],data['y_train'],data['y_test']\n",
+    "x_train = x_train[:,1:]\n",
+    "x_test = x_test[:,1:]\n",
+    "# x_valid, x_test = x_test[0:2880,:], x_test[2880:,:]\n",
+    "# y_valid, y_test = y_test[0:2880], y_test[2880:]\n",
+    "print(x_train.shape,x_test.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 8 features: HolidayFlag, DayOfWeek, WeekOfYear, Month, ForecastWindProduction, Forecasted SystemLoad,  Forecasted Energy price, Forecasted CO2Intensity\n",
+    "#### And the predictor variable: Energy Price"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### First we standardize the feature variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scaler = StandardScaler()\n",
+    "x_train = scaler.fit_transform(x_train)\n",
+    "x_test = scaler.transform(x_test)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### We will use this data in the context of knapsack problem. The predicted  Energy Price forms the value of knapsack. In our setup, each knapsack instance consists of 48 items. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "240\n"
+     ]
+    }
+   ],
+   "source": [
+    "weights = [data['weights'].tolist()]\n",
+    "weights = np.array(weights)\n",
+    "n_items = 48\n",
+    "capacity = 60\n",
+    "print(np.sum(weights))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### The weights are the costs of each item. The weights are kept same for each instance. The summation of 48 weights are 240. The knapsack capacity in this example is 60"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### We reshape the data in groups of 48 so that each group is an instance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(552, 48, 8) (237, 48, 8)\n"
+     ]
+    }
+   ],
+   "source": [
+    "x_train = x_train.reshape(-1,48,x_train.shape[1])\n",
+    "y_train = y_train.reshape(-1,48)\n",
+    "x_test = x_test.reshape(-1,48,x_test.shape[1])\n",
+    "y_test = y_test.reshape(-1,48)\n",
+    "print(x_train.shape,x_test.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### As we see, we have 552 training instances and 237 test instances"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Next we wil randomize the training and test data"
```

Comment:
this can be part of an earlies cell perhaps, with a comment; I thnk it could be good to have a bit fewer boxes...